source/MaterialXCore/Interface.cpp

```diff
@@ -339,6 +359,7 @@ bool Input::validate(string* message) const
     {
         bool hasValueBinding = hasValue();
         bool hasConnection = hasNodeName() || hasNodeGraphString() || hasOutputString() || hasInterfaceName();
+        validateRequire(!(hasValueBinding && hasConnection), res, message, "Node input binds to both a value and a connection");
```

Comment:
What about an input that has both `interfacename`  and `nodename`? Should really be "only one of". You can do the combinatorial, or use an integer:
```C++
int numBindings = 0;
if (hasValue()) numBindings ++;
if (hasNodeName()) numBindings ++;
if (hasNodeGraphStrings()) numBindings ++;
if (hasInterfaceName()) numBindings ++;
if (hasOutputString() && !(hasNodeName() || hasNodeGraphString()))  numBindings ++;
validateRequire(numBindings, res, message, "Node input binds no value or connection");
validateRequire(numBindings <= 1, res, message, "Node input has too many bindings");
```

Reply:
Thanks for the logic! I've replaced it with your code and added in additional attribute removes.